src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,152 @@
+        ):
+            aggregated_projects = defaultdict(list)
+            start_time = time.time()
+            offset = 0
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                query = (
+                    Query(
+                        dataset=Dataset.Metrics.value,
+                        match=Entity(EntityKey.OrgMetricsCounters.value),
+                        select=[
+                            Column("org_id"),
+                            Column("project_id"),
+                        ],
+                        groupby=[Column("org_id"), Column("project_id")],
+                        where=[
+                            Condition(
+                                Column("timestamp"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
+                            ),
+                            Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                            Condition(
+                                Column("metric_id"),
+                                Op.EQ,
+                                SESSION_METRIC_NAMES["c:sessions/session@none"],
+                            ),
+                        ],
+                        granularity=Granularity(3600),
+                        orderby=[
+                            OrderBy(Column("org_id"), Direction.ASC),
+                            OrderBy(Column("project_id"), Direction.ASC),
+                        ],
+                    )
+                    .set_limit(self.CHUNK_SIZE + 1)
+                    .set_offset(offset)
+                )
+                data = raw_snql_query(
+                    query, referrer="release_monitor.fetch_projects_with_recent_sessions"
+                )["data"]
+                count = len(data)
+                more_results = count > self.CHUNK_SIZE
+                offset += self.CHUNK_SIZE
+
+                if more_results:
+                    data = data[:-1]
+
+                for row in data:
+                    aggregated_projects[row["org_id"]].append(row["project_id"])
+
+                if not more_results:
+                    break
+
+            else:
+                logger.error(
+                    "release_monitor.fetch_projects_with_recent_sessions.loop_timeout",
+                    extra={"offset": offset},
+                )
+
+        return aggregated_projects
+
+    def fetch_project_release_health_totals(
+        self, org_id: int, project_ids: Sequence[int]
+    ) -> Totals:
+        start_time = time.time()
+        offset = 0
+        totals: Totals = defaultdict(dict)
+        with metrics.timer("release_monitor.fetch_project_release_health_totals.loop"):
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                with metrics.timer("release_monitor.fetch_project_release_health_totals.query"):
+                    query = (
+                        Query(
```

Comment:
This query can go through the metrics layer for sure.. See https://github.com/getsentry/sentry/pull/33735 to construct a `QueryDefinition` object that the metrics func here can take https://github.com/getsentry/sentry/pull/33735/files#diff-fe042572eb7930f0cd4d0ce5b501b92fe1fd200e29a24b86004245745d8e8812R423
cc @jjbayer Can you fact check me on that?